src/util.c

```diff
@@ -645,8 +645,26 @@ printsocket(struct tcb *tcp, int fd, const char *path)
 	return true;
 }
 
+typedef bool (*scan_fdinfo_fn)(const char *value, void *data);
+
+static bool
+scan_fdinfo(pid_t pid_of_fd, int fd, const char *search_pfx,
+	    size_t search_pfx_len, scan_fdinfo_fn fn, void *data);
```

Comment:
The last question is whether we could rearrange some code to avoid these forward declarations.

Reply:
Rearranged.